backend/src/ws.rs

```diff
@@ -48,32 +59,40 @@ pub async fn client_connection(ws: WebSocket, clients: Clients) {
                 break;
             }
         };
-        client_msg(&uuid, msg, &clients).await;
+        client_msg(&uuid, msg).await;
     }
 
-    // as the above will keep running as long as the client is active, when we exit the loop we can safely remove this client instance from the hashmap.
-    clients.lock().await.remove(&uuid);
-    println!("{} disconnected", uuid); //debug
+    // as the above will keep running as long as the client is active, when we exit the loop we can safely remove this client instance from the hashmap, after we have removed its position from the grid.
+    call_remove_player(&uuid).await;
+    println!("{uuid} disconnected"); //debug
 }
 
+async fn add_player(uuid: String, new_client: Client) {
+    CLIENTS.lock().await.insert(uuid, new_client);
+}
+
+async fn call_remove_player(uuid: &str) {
+    remove_player(CLIENTS.lock().await.get(uuid).unwrap().last_position).await;
+    CLIENTS.lock().await.remove(uuid);
+}
 
 // ->recieve client game info <- send back client game state
 // wwwwwwwwwwwwwwwwwwwww i am so tired
-async fn client_msg(client_id: &str, msg: Message, clients: &Clients) {
+async fn client_msg(client_id: &str, msg: Message) {
     //println!("received message from {}: {:?}", client_id, msg); //debug
 
     let message = match msg.to_str() {
         Ok(v) => v,
         Err(_) => return,
     };
 
-    //println!("{}", message);
+    //println!("{message}");
 
-    let locked = clients.lock().await;
+    let locked = CLIENTS.lock().await;
     match locked.get(client_id) {
         Some(v) => {
             if let Some(sender) = &v.sender {
-                let _ = sender.send(Ok(Message::text(create_response(message))));
+                let _ = sender.send(Ok(Message::text(create_response(message, client_id).await)));
             }
         }
         None => return,
```

Comment:
Can you add a println for this error state plx.

Reply:
Should now be done in b0f5c6126921c7de45e237b425c08414ca23f125